docs/current_docs/api/fs-filters.mdx

```diff
@@ -0,0 +1,308 @@
+```typescript
+// exclude tests and test data
+@argument({ ignore: ["**_test.go", "**/testdata/**"] })
+
+// exclude binaries
+@argument({ ignore: ["bin"] })
+
+// exclude Python dependencies
+@argument({ ignore: ["**/.venv","**/__pycache__"] })
+
+// exclude Node.js dependencies
+@argument({ ignore: ["**/node_modules"] })
+
+// exclude Git metadata
+@argument({ ignore: ["*", "!.git", "!**/.gitignore"] })
+```
+</TabItem>
+</Tabs>
+
+## Post-call filtering
+
+Post-call filtering means that a directory is filtered after it's uploaded to the Dagger Engine.
+
+This is useful when working with directories that are modified "in place" by a Dagger Function. When building an application, your Dagger Function might modify the source directory during the build by adding new files to it. A post-call filter allows you to use that directory in another operation, only fetching the new files and ignoring the old ones.
+
+A good example of this is a multi-stage build. Imagine a Dagger Function that reads and builds an application from source, placing the compiled binaries in a new sub-directory (stage 1). Instead of then transferring everything to the final container image for distribution (stage 2), you could use a post-call filter to transfer only the compiled files.
+
+<Tabs groupId="language">
+<TabItem value="Go">
+
+To implement a post-call filter in your Dagger Function, use the `DirectoryWithDirectoryOpts` or `ContainerWithDirectoryOpts` structs, which support `Include` and `Exclude` patterns for `Directory` objects. Here's an example:
+
+```go file=./snippets/fs-filters/post-call/go/main.go
+```
+
+</TabItem>
+<TabItem value="Python">
+
+To implement a post-call filter in your Dagger Function, use the `include` and `exclude` parameters when working with `Directory` objects. Here's an example:
+
+```python file=./snippets/fs-filters/post-call/python/main.py
+```
+
+</TabItem>
+<TabItem value="TypeScript">
+
+To implement a post-call filter in your Dagger Function, use the `include` and `exclude` parameters when working with `Directory` objects. Here's an example:
+
+```typescript file=./snippets/fs-filters/post-call/typescript/index.ts
+```
+
+</TabItem>
+</Tabs>
+
+Here are a few examples of useful patterns:
+
+<Tabs groupId="language">
+<TabItem value="Go">
+```go
+// exclude all Markdown files
+dirOpts := dagger.ContainerWithDirectoryOpts{
+  Exclude: "*.md*",
+}
+
+// include only the build output directory
+dirOpts := dagger.ContainerWithDirectoryOpts{
+  Include: "build",
+}
+
+// include only ZIP files
+dirOpts := dagger.DirectoryWithDirectoryOpts{
+  Include: "*.zip",
+}
+
+// exclude Git metadata
+dirOpts := dagger.DirectoryWithDirectoryOpts{
+  Exclude: "*.git",
+}
+```
+</TabItem>
+<TabItem value="Python">
+```python
+# exclude all Markdown files
+include=["*.md*"]
+
+# include only the build output directory
+include=["build"]
+
+# include only ZIP files
+include=["*.zip"]
+
+# exclude Git metadata
+exclude=["*.git"]
```

Comment:
dirOpts is missing here? Is it on purpose?

Reply:
Updated